src/structs.jl

```diff
@@ -50,3 +50,173 @@ IPUMS.IPUMSSource("IPUMS USA", "usa", "microdata", "", true, "")
+    varinterval = "discrete"
+    )
+
+IPUMS.DDIVariable("YEAR", 1, 4, 4, "Survey year", "YEAR reports the year in which the survey was conducted.  YEARP is repeated on person records.", 0, "numeric", "discrete")
+DDIVariable("IPUMS USA", "usa", "microdata", "", true, "")
+```
+
+# References
+
+Information about each variable field is taken from: 
+
+https://ddialliance.org/Specification/DDI-Codebook/2.5/XMLSchema/field_level_documentation_files/schemas/codebook_xsd/elements/stdyDscr.html
+"""
+@kwdef mutable struct DDIVariable
+    name::String = ""
+    position_start::Int64 = 9999
+    position_end::Int64 = 9999
+    position_width::Int64 = 9999 
+    labl::String = ""
+    desc::String = ""
+    dcml::Int64 = 9999
+    vartype::String = ""
+    varinterval::String = ""
+end
+
+
+"""
+```julia
+DDIInfo(
+    filepath::String, 
+    conditions::String = "", 
+    citation::String = "", 
+    ipums_project::String = "",
+    extract_notes::Sring = "", 
+    extract_date::String = "",
+    variable_info::Vector{DDIVariable} = DDIVariable[]
+    _xml_doc::EzXML.Document = EzXML.XMLDocument()
+    _ns::String = ""
+)
+```
+
+A struct representing the metadata taken from an IPUMs extract. An IPUMs 
+extract contains both file-level metadata--such as the date of export--as well
+as variable level metadata--such as the name and data type of a variable. 
+
+The `DDIInfo` object is not generally called directly. The `parse_ddi()` 
+function creates a `DDIinfo` object after successfully parsing a DDI 
+file from an IPUMs extract. 
+
+The `DDIInfo`
+object contain file level metadata. The `variable_info` field of the `DDIInfo` 
+object contains a vector of `DDIVariable` objects. `DDIVariable` objects contain
+metadata information about individual IPUMs variables.   
+
+# Keyword Arguments
+
+- `filepath::String` -  File system path to the DDI (`.xml``) file.
+- `conditions::String` - IPUMs legal specification on the proper use of IPUMs 
+                            data.
+- `citation::String` - Information for the citation of IPUMs data.
+- `ipums_project::String` - Identifier for the IPUMs source of the extract 
+                            data, such as `IPUMS CPS`, or `IPUMS USA`, etc. 
+- `extract_notes::String` - Additional clarifying information or user nodes 
+                            about the extract. 
+- `extract_date::String` - Date on which the extract was produced.
+- `variable_info::Vector{DDIVariable}` - a vector of `DDIVariable` objects,
+                                        which contain metadata on each variable
+                                        or column in the data file.
+- `_xml_doc::EzXML.Document` - An internal attribute that contains an internal 
+                                representation of the DDI DOM for parsing.
+- `_ns::String` - An internal attribute to hold any namespaces used in the 
+                    XML DOM.
+
+# Returns
+
+- `DDIInfo` object that contains both file-level and variable-level metadata
```

Comment:
I am unsure why but could you put these two lines on the same line? Otherwise, the Julia REPL seems not to render this properly.

Reply:
Okay, should be fixed as per your instruction.